cgi-bin/DW/Setting/StickyEntry.pm

```diff
@@ -48,11 +70,39 @@ sub option {
 sub save {
     my ( $class, $u, $args ) = @_;
 
-    my $sticky = $class->get_arg( $args, "stickyid" ) || '';
-    $sticky = LJ::text_trim( $sticky, 0, 100 );
-    unless ( $u->sticky_entry ( $sticky ) ) {
-        $class->errors( "stickyid" => $class->ml( 'setting.stickyentry.error.invalid' ) ) ;
+    my $max_sticky_count = $u->count_max_stickies;
+    my @stickies;
+    # Create a hash that we will use to check for duplicate entries.
+    my %unique = ();
+    my $username = $u->user;
+
+    for my $i ( 1 ... $max_sticky_count ) {
+        my $stickyi = $class->get_arg( $args, "stickyid${i}" ) || '';
+
+        # blank form entry
+        next unless $stickyi;
+
+        $stickyi = LJ::text_trim( $stickyi, 0, 100 );
+        my $e = LJ::Entry->new_from_url_or_ditemid( $stickyi, $u );
+        if ( $e ) {
+            my $ditemid = $e->ditemid;
+            if ( $unique{$ditemid} ) {
+                $class->errors( "stickyid" => ( $class->ml( 'setting.stickyentry.error.duplicate', { stickyid => $i } ) ) ) ;
+                return 1;
+            }
+            push @stickies,  $ditemid;
+            $unique{$ditemid} = 1;
+        } else {
+            # As soon as we detect a problem with a sticky we break out of the subroutine.
+            $class->errors( "stickyid" => ( $class->ml( 'setting.stickyentry.error.invalid2', { stickyid => $i } ) ) ) ;
+            return 1;
+        }
     }
+
+    my $remote = LJ::get_remote();
```

Comment:
Authorization is done at the _page_ level, not in a setting.

So, this needs to be in the controller. I don't think authorization should go anywhere else.